components/bl-ace-editor-component/src/index.js

```diff
@@ -0,0 +1,61 @@
+import { useState, useEffect, useCallback } from 'react';
+
+import { AceEditor } from './lib/react-ace.umd';
+import './lib/ace-builds.umd';
+
+const { cn } = BackendlessUI.CSSUtils;
+
+export default function AceEditorComponent({ component, elRef, eventHandlers }) {
+  const {
+    classList, display, style, disabled, name, value, mode, theme,
+    foldStyle, placeholder, width, height, fontSize, tabSize, printMarginColumn,
+    printMarginVisibility, gutterVisibility, autocompletion, highlightActiveLine, showInvisibles
+  } = component;
+  const { onChange } = eventHandlers;
+
+  const [editorValue, setEditorValue] = useState('');
+
+  component.getValue = () => editorValue;
+  component.setValue = value => setEditorValue(value);
+
+  useEffect(() => {
+    setEditorValue(value);
+  }, [value]);
+
+  const handleValueChange = useCallback((newValue) => {
+    setEditorValue(newValue);
+    onChange({ value: newValue });
+  }, []);
+
+  if (!display) {
+    return null;
+  }
+
+  return (
+    <div ref={ elRef } className={ cn("bl-customComponent-aceEditor", classList) } style={ style }>
+      <AceEditor
+        value={ editorValue }
+        mode={ mode }
+        theme={ theme }
+        width={ width }
+        height={ height }
+        readOnly={ disabled }
```

Comment:
> The difference between disabled and readonly is that read-only controls can still function and are still focusable, whereas disabled controls can not receive focus and are not submitted with the form and generally do not function as controls until they are enabled.

Is it really disabled? May be we need to use same definition as library(readonly)?